reducers/plugins/index.js

```diff
@@ -27,6 +27,17 @@ function removePluginComponents(state, action) {
     return state;
 }
 
+function removeMainMenuAction(state, action) {
+    if (!action.data) {
+        return state;
+    }
+
+    let nextState = [...state];
+    nextState = nextState.filter((item) => item.id !== action.data.id);
```

Comment:
`filter` already returns a copy of the array -- any particular reason to clone it on line 35?

Reply:
Nope, I was originally using a different loop type where I needed the copy. Will update